### PR TITLE
STORM-2847 1.x

### PR DESCRIPTION
**Pull request by srdo**
1.x for https://github.com/apache/storm/pull/2454

### CODE REVIEW

File: external/storm-kafka-client/src/test/java/org/apache/storm/kafka/spout/KafkaSpoutReactivationTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                KafkaSpoutConfig.builder("127.0.0.1:" + kafkaUnitRule.getKafkaUnit().getKafkaPort(),
+                    SingleTopicKafkaSpoutConfiguration.TOPIC))
+                .setFirstPollOffsetStrategy(UNCOMMITTED_EARLIEST)
+                .setOffsetCommitPeriodMs(commitOffsetPeriodMs)
+                .setProp(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, maxPollRecords)
+                .build();
+        KafkaConsumerFactory<String, String> consumerFactory = new KafkaConsumerFactoryDefault<>();
+        this.consumerSpy = spy(consumerFactory.createConsumer(spoutConfig));
+        this.postReactivationConsumerSpy = spy(consumerFactory.createConsumer(spoutConfig));
+        KafkaConsumerFactory<String, String> consumerFactoryMock = mock(KafkaConsumerFactory.class);
```
Comment (hmcl) [inline external/storm-kafka-client/src/test/java/org/apache/storm/kafka/spout/KafkaSpoutReactivationTest.java]: why do you need consumerFactory and consumerFactoryMock?

Comment (srdo) [reply external/storm-kafka-client/src/test/java/org/apache/storm/kafka/spout/KafkaSpoutReactivationTest.java]: I need to verify that all messages were committed https://github.com/apache/storm/pull/2464/files#diff-ed317082b62537f44916bda817cf1cecR120 which is done by checking if commit was called on the consumer spy.

**Comment by hmcl** [comment]
Can you please squash the commits. You can merge after.

**Comment by hmcl** [comment]
@srdo do you think we can get this PR and https://github.com/apache/storm/pull/2465 merged in today? Thanks.

**Comment by srdo** [comment]
@hmcl Yes, I'll squash and merge this now. Could you make sure that https://github.com/apache/storm/pull/2466 is up to date with the 1.x version, then I'll do a quick review of that?

**Comment by hmcl** [comment]
@srdo OK, I am wrapping up #2466. Hopefully we can still get this done today. Thanks.

**Pull request merged**